src/main/java/strategy/SupplyOperation.java

```
@@ -0,0 +1,11 @@
+package strategy;
+
+import db.Storage;
+import model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
```

This violates the requirement to anticipate invalid inputs (see checklist examples: "...for example: 1. Incorrect file path 2. Incorrect data in the input file, for example, quantity less than zero or incorrect strategy 3. Null parameters"). The `apply(FruitTransaction transaction)` method does not check whether `transaction` is null before using it. Add a null-check at the start of the method (for example: `if (transaction == null) throw new IllegalArgumentException("Transaction cannot be null");`) so the handler fails fast with an informative message instead of causing an NPE later. See checklist for the exact examples .